lld/utils/speed-test-reproducers/collect.nix

```diff
@@ -0,0 +1,157 @@
+        clangStdenv = reproducerCollectingStdenv pkgs.clangStdenv pkgs.buildPackages.lld;
+      };
+    in
+    {
+      # For benchmarking the linker we want to disable LTO as otherwise we would
+      # just be benchmarking the LLVM optimizer. Also, we generally want the
+      # package to use the regular stdenv in order to simplify wrapping it.
+      # Firefox normally uses the rustc stdenv but uses the regular one if
+      # LTO is disabled so we kill two birds with one stone by disabling it.
+      # Chromium uses the rustc stdenv unconditionally so we need the stuff
+      # below to make sure that it finds our wrapped stdenv.
+      chrome =
+        (pkgs.chromium.override {
+          newScope =
+            extra:
+            pkgs.newScope (
+              extra
+              // {
+                pkgsBuildBuild = {
+                  pkg-config = pkgs.pkgsBuildBuild.pkg-config;
+                  rustc = {
+                    llvmPackages = rec {
+                      stdenv = reproducerCollectingStdenv pkgs.pkgsBuildBuild.rustc.llvmPackages.stdenv pkgs.pkgsBuildBuild.rustc.llvmPackages.lld;
+                      bintools = stdenv.cc.bintools;
+                    };
+                  };
+                };
+              }
+            );
+          pkgs = {
+            rustc = {
+              llvmPackages = {
+                stdenv = reproducerCollectingStdenv pkgs.rustc.llvmPackages.stdenv pkgs.rustc.llvmPackages.lld;
+              };
+            };
+          };
+        }).browser.overrideAttrs
+          (old: {
+            configurePhase =
+              old.configurePhase
+              + ''
+                echo use_thin_lto = false >> out/Release/args.gn
+                echo is_cfi = false >> out/Release/args.gn
+              '';
+          });
+      firefox = (withReproducerCollectingStdenv pkgs.firefox-unwrapped).override {
+        ltoSupport = false;
+        pgoSupport = false;
+      };
+      # Won't work until https://github.com/NixOS/nixpkgs/pull/390631 lands.
+      # Can replace above line with
+      #   nixpkgsDir ? fetchTarball "https://github.com/NixOS/nixpkgs/archive/fbc5923fb30c7e1957a729f19f22968083fb473f.tar.gz",
+      # for testing with that PR.
+      linux-kernel = (withReproducerCollectingStdenv pkgs.linux_latest).dev;
+      ladybird = withReproducerCollectingStdenv pkgs.ladybird;
+      llvm = withReproducerCollectingStdenv pkgs.llvm;
+      webkitgtk = withReproducerCollectingClangStdenv pkgs.webkitgtk;
+      hello = withReproducerCollectingStdenv pkgs.hello;
+    };
+    targets = {
+      x86_64 = reproducerPkgs { config = "x86_64-unknown-linux-gnu"; };
+      aarch64 = reproducerPkgs { config = "aarch64-unknown-linux-gnu"; };
+      riscv64 = reproducerPkgs { config = "riscv64-unknown-linux-gnu"; };
+    };
+    nativePkgs = nixpkgs { };
+in
+derivation {
+  name = "lld-speed-test";
+  system = builtins.currentSystem;
```

Comment:
Please make this an option in the attrs argument. You can have it be `system ? builtins.currentSystem`.